WeakAuras/AuraEnvironment.lua

```diff
@@ -388,7 +455,10 @@ local FakeWeakAurasMixin = {
   },
   override = {
     me = GetUnitName("player", true),
-    myGUID = UnitGUID("player")
+    myGUID = UnitGUID("player"),
+    GetData = function(id)
+      return CopyTable(WeakAuras.GetData(id))
```

Comment:
there's a perf risk here if an aura calls GetData many times. since they're not supposed to do that idk how much i care, but using a cache could be an enhancement if it ends up being a problem users complain too loudly about

```suggestion
      getDataCache = setmetatable({}, {__index = function(self, id) local copy = Copytable(WeakAuras.GetData) self[id] = copy return copy end})
      return getDataCache[id]
```

Reply:
I actually didn't implement a cache after all. Instead I added an aura warning if an aura calls GetData >= 100 times since the last init. That should catch auras that don't cache the call.

I tested it. btw, and CopyTable is pretty fast, so calling it repeately isn't actually that bad. Thus we can eventually remove the warning since the main prupose is just to warn users that relied on the function being essentially free.